src/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs

```diff
@@ -0,0 +1,226 @@
+			var distributedTracingData = ExtractIncomingDistributedTracingData(httpRequest);
+			if (distributedTracingData != null)
+			{
+				_logger.Debug()
+					?.Log(
+						"Incoming request with {TraceParentHeaderName} header. DistributedTracingData: {DistributedTracingData} - continuing trace",
+						TraceParent.TraceParentHeaderName, distributedTracingData);
+
+				_currentTransaction = Agent.Instance.TracerInternal.StartTransactionInternal(
+					$"{httpRequest.HttpMethod} {httpRequest.Path}",
+					ApiConstants.TypeRequest,
+					distributedTracingData);
+			}
+			else
+			{
+				_logger.Debug()?.Log("Incoming request doesn't have valid incoming distributed tracing data - starting trace with new trace id.");
+				_currentTransaction = Agent.Instance.TracerInternal.StartTransactionInternal(
+					$"{httpRequest.HttpMethod} {httpRequest.Path}",
+					ApiConstants.TypeRequest);
+			}
+
+			if (_currentTransaction.IsSampled) FillSampledTransactionContextRequest(httpRequest, _currentTransaction);
+		}
+
+		private static DistributedTracingData ExtractIncomingDistributedTracingData(HttpRequest httpRequest)
+		{
+			var headerValue = httpRequest.Headers.Get(TraceParent.TraceParentHeaderName);
+			if (headerValue == null)
+			{
+				_logger.Debug()?.Log("Incoming request doesn't {TraceParentHeaderName} header - "+
+					"it means request doesn't have incoming distributed tracing data", TraceParent.TraceParentHeaderName);
+				return null;
+			}
+			return TraceParent.TryExtractTraceparent(headerValue);
+		}
+
+		private static void FillSampledTransactionContextRequest(HttpRequest httpRequest, ITransaction transaction)
+		{
+			var httpRequestUrl = httpRequest.Url;
+			var url = new Url
+			{
+				Full = httpRequestUrl.AbsoluteUri,
+				HostName = httpRequestUrl.Host,
+				Protocol = "HTTP",
+				Raw = httpRequestUrl.OriginalString
+			};
+
+			transaction.Context.Request = new Request(httpRequest.HttpMethod, url)
+			{
+				Socket = new Socket
+				{
+					Encrypted = httpRequest.IsSecureConnection,
+					RemoteAddress = httpRequest.UserHostAddress
+				},
+				HttpVersion = GetHttpVersion(httpRequest.ServerVariables["SERVER_PROTOCOL"]),
+				Headers = _isCaptureHeadersEnabled ? ConvertHeaders(httpRequest.Headers) : null
+			};
+		}
+
+		private static string GetHttpVersion(string protocolString)
+		{
+			switch (protocolString)
+			{
+				case "HTTP/1.0":
+					return "1.0";
+				case "HTTP/1.1":
+					return "1.1";
+				case "HTTP/2.0":
+					return "2.0";
+				default:
+					return protocolString.Replace("HTTP/", string.Empty);
+			}
+		}
+
+		private static Dictionary<string, string> ConvertHeaders(NameValueCollection httpHeaders)
+		{
+			var convertedHeaders = new Dictionary<string, string>();
+			foreach (var headerName in httpHeaders.AllKeys)
+				convertedHeaders.Add(headerName, string.Join(",", httpHeaders.GetValues(headerName)));
+			return convertedHeaders;
+		}
+
+		private void ProcessEndRequest(object eventSender, EventArgs eventArgs)
```

Comment:
nit: Same as the ProcessBeginRequest, `eventArgs` is unused.

Reply:
Done.